src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Lower.java

```diff
@@ -126,6 +128,10 @@ protected Lower(Context context) {
             target.optimizeOuterThis() ||
             options.getBoolean("optimizeOuterThis", false);
         disableProtectedAccessors = options.isSet("disableProtectedAccessors");
+        Source source = Source.instance(context);
```

Comment:
Does this merit addition of a feature-enum constant in the usual way?

Reply:
I was thinking of this, but opted to reuse the existing feature constant, as this is not a separate user-selectable feature, but rather part of the patterns in switch feature. Also, each feature enum constant must a diagnostic set, and as it would never be part of any diagnostic, it would be impossible to test.

It is possible to have a separate constant if desired, though.

Reply:
I'm OK with the current approach